x-pack/plugins/aiops/public/components/change_point_detection/change_points_table.tsx

```diff
@@ -0,0 +1,364 @@
+      render: (annotation: ChangePointAnnotation) => {
+        return <MiniChartPreview annotation={annotation} fieldConfig={fieldConfig} />;
+      },
+    },
+    {
+      field: 'type',
+      name: i18n.translate('xpack.aiops.changePointDetection.typeColumn', {
+        defaultMessage: 'Type',
+      }),
+      sortable: true,
+      truncateText: false,
+      render: (type: ChangePointAnnotation['type']) => <EuiBadge color="hollow">{type}</EuiBadge>,
+    },
+    {
+      field: 'p_value',
+      name: i18n.translate(
+        'xpack.aiops.explainLogRateSpikes.spikeAnalysisTableGroups.pValueLabel',
+        {
+          defaultMessage: 'p-value',
+        }
+      ),
+      sortable: true,
+      truncateText: false,
+      render: (pValue: ChangePointAnnotation['p_value']) => pValue.toPrecision(3),
+    },
+    ...(fieldConfig.splitField
+      ? [
+          {
+            field: 'group.name',
+            name: i18n.translate('xpack.aiops.changePointDetection.fieldNameColumn', {
+              defaultMessage: 'Field name',
+            }),
+            truncateText: false,
+          },
+          {
+            field: 'group.value',
+            name: i18n.translate('xpack.aiops.changePointDetection.fieldValueColumn', {
+              defaultMessage: 'Field value',
+            }),
+            truncateText: false,
+            sortable: true,
+          },
+        ]
+      : []),
+  ];
+
+  const selectionValue = useMemo<EuiTableSelectionType<ChangePointAnnotation>>(() => {
+    return {
+      selectable: (item) => true,
+      onSelectionChange: (selection) => {
+        onSelectionChange(
+          selection.map((s) => {
+            return {
+              ...s,
+              ...fieldConfig,
+            };
+          })
+        );
+      },
+    };
+  }, [fieldConfig, onSelectionChange]);
+
+  return (
+    <EuiInMemoryTable<ChangePointAnnotation>
+      itemId={'id'}
+      selection={selectionValue}
+      loading={isLoading}
+      items={annotations}
+      columns={columns}
+      pagination={{ pageSizeOptions: [5, 10, 15] }}
+      sorting={sorting}
+      message={
+        <EuiEmptyPrompt
+          iconType="search"
+          title={
+            <h2>
+              <FormattedMessage
+                id="xpack.aiops.changePointDetection.noChangePointsFoundTitle"
+                defaultMessage="No change points found"
+              />
+            </h2>
+          }
+          body={
+            <p>
+              <FormattedMessage
+                id="xpack.aiops.changePointDetection.noChangePointsFoundMessage"
+                defaultMessage="Detect statistically significant change points such as dips, spikes, and distribution changes in a metric. Select a metric and set a time range to start detecting change points in your data."
+              />
+            </p>
+          }
+        />
+      }
+    />
+  );
+};
+
+export const MiniChartPreview: FC<ChartComponentProps> = ({ fieldConfig, annotation }) => {
```

Comment:
Should this MiniChartPreview be in its own file?

Reply:
I reckon it's fine to keep several components in one file when they used together exclusively.